### ISSUE TITLE
Multiple tracks

### ISSUE THREAD

**Issue by madclam** [codalab/codalab-competitions#816]
There is a limitation of Codalab presently that forces users to submit 1 entry for all the tasks when we have multiple datasets/tasks in a given phase. For example, in each phase of the AutoML challenge we have 5 tasks. Therefore, the computational time limit is globally computed on the 5 tasks. 

What we really meant to have is a time limit (which is different) for each task. At present the global limit is the sum of the individual task limits and it is up to the participants to manage their time. But we cannot check how they do their time management. Therefore, it is possible for them to re-allocate time differently.

We talked in the past of having parallel tracks within a phase. One solution would be to launch multiple processes with the metadata file when you make a submission as a participant.

For instance, now a submission must include in the metadata file:
command: python $program/run.py $input $output

Could we have:
command: python $program/run.py $input/set1 $output/set1
command: python $program/run.py $input/set2 $output/set2
command: python $program/run.py $input/set3 $output/set3
command: python $program/run.py $input/set4 $output/set4
command: python $program/run.py $input/set5 $output/set5

that is 5 separate commands for the 5 datasets.

I think this would solve the problem. Are there any possible potential side effects?

**Comment by ckcollab** [comment]
Save `elapsedTime`/metadata into the $output/sub_dir

**Comment by madclam** [comment]
I would like to add that we should leave open the possibility of executing in this way also multiple jobs that are executed in sequence, one feeding its output to the next one, instead of being independent jobs:
command: python $program/run1.py $input $output/subdir1
command: python $program/run2.py $output/subdir1 $output/subdir2
command: python $program/run3.py $output/subdir2 $output/subdir3

The question is: how do we create the appropriate directory structure. For the log files, we can concatenate all the logs.

**Comment by ErickWa** [comment]
Isabelle's test failed, so we need further debugging

**Comment by ErickWa** [comment]
Hello,
recently I was testing the 'multiple track' feature. https://github.com/codalab/codalab/issues/816
The goal was to execute 5 datasets separately.
Debugging it was very hard. I have been submitting, but it was not running.
I tried with just one dataset, like:
command: python $program/run.py $input/adult $output

It worked. However, the stdout/err were empty (seems like if is saved sometimes and last part before error is lost)
Problem 1) The stdout/err was empty or partial even though it worked with one dataset.

I used the commands from Isabelle, but there was a typo there.
I didn't know it and there was no feedback that the folder does not exist, also due to 1)
Why then the first datasets were not calculated? It seems that only the last line is executed, so even though there are 5 lines, only the last one is executed (or is executed but their outputs are overwritten and this is why the only last one is kept) -- problem 2). 

Unfortunately, the last dataset name was with this typo, so the whole submission was not executed at all and none stderr was not stored. Maybe we can add info what's executed (e.g. append executed command to stderr, problem 3)).


Best,
Lukasz
